EventCatalogAPI/Controllers/EventItemsController.cs

```diff
@@ -63,5 +68,34 @@ public async Task<IActionResult> Dates()
             return Ok(events);
         }
 
+        //List of Event Categories
+        [HttpGet]
+        [Route("[action]")]
+
+        public async Task<IActionResult> CategoryLists()
+        {
+            var events = await _context.EventCategory.ToListAsync();
+            return Ok(events);
+        }
+
+
+        //Sort Event by Category 
+        [HttpGet]
+        [Route("[action]")]
+        public async Task<IActionResult> EventCategories(
+            [FromQuery] int pageIndex = 0,
+            [FromQuery] int pageSize = 4)
+        {
+
+            var events = await _context.EventItems
+                    .OrderBy(c => c.EventCategory)
```

Comment:
We need to add a Where statement before the OrderBy using the eventCategory we got from the query - something like this: ".Where(c => c.EventCategory == category)"